zerver/views/auth.py

```diff
@@ -191,6 +194,13 @@ def maybe_send_to_registration(
         set_expirable_session_var(
             request.session, "registration_desktop_flow_otp", desktop_flow_otp, expiry_seconds=3600
         )
+        if params_to_store_in_authenticated_session:
+            set_expirable_session_var(
+                request.session,
+                "registration_desktop_flow_params_to_store_in_authenticated_session",
+                orjson.dumps(params_to_store_in_authenticated_session).decode(),
+                expiry_seconds=3600,
+            )
```

Comment:
Should we extract the 3600 as a constant?

Also, is there any possible action where you do an authentication with SAML SLO enabled, and get `registration_desktop_flow_params_to_store_in_authenticated_session` set, but then do a second authentication without it, and be able to use the parameters from a previous authentication attempt with a new user/credentials. It's not clear there's anything bad possible, since you've shown you control both accounts... but basically, I'm wondering if there should be an "else clear `registration_desktop_flow_params_to_store_in_authenticated_session`" block here just in case.

Reply:
Extracted in a prep commit.

Hmm, I'm not sure if I understand exactly the scenario you have in mind - but if you're doing authentication the second time, then I guess you mean you logged out? And upon logging out, the session gets cleared:
(https://docs.djangoproject.com/en/4.2/topics/auth/default/#django.contrib.auth.logout)

> When you call [logout()](https://docs.djangoproject.com/en/4.2/topics/auth/default/#django.contrib.auth.logout), the session data for the current request is completely cleaned out. All existing data is removed. This is to prevent another person from using the same web browser to log in and have access to the previous user’s session data.